beacon-chain/state/testing/getters.go

```diff
@@ -0,0 +1,192 @@
+package testing
+
+import (
+	"sync"
+	"testing"
+
+	types "github.com/prysmaticlabs/eth2-types"
+	"github.com/prysmaticlabs/prysm/beacon-chain/state"
+	fieldparams "github.com/prysmaticlabs/prysm/config/fieldparams"
+	ethpb "github.com/prysmaticlabs/prysm/proto/prysm/v1alpha1"
+	"github.com/prysmaticlabs/prysm/testing/assert"
+	"github.com/prysmaticlabs/prysm/testing/require"
+)
+
+func VerifyBeaconState_SlotDataRace(t *testing.T, factory getState) {
+	headState, err := factory()
+	require.NoError(t, err)
+
+	wg := sync.WaitGroup{}
+	wg.Add(2)
+	go func() {
+		require.NoError(t, headState.SetSlot(0))
+		wg.Done()
+	}()
+	go func() {
+		headState.Slot()
+		wg.Done()
+	}()
+
+	wg.Wait()
+}
+
+type getStateWithCurrentJustifiedCheckpoint func(*ethpb.Checkpoint) (state.BeaconState, error)
```

Comment:
Maybe a less verbose name here ? ex: `jStateRetriever` or `justifiedStateRetriever`

Reply:
I wouldn't know that j means justified, and the similar methods start with getState as suggested by @terencechain before as it would follow more the rest of the code.

Also, looking at the name of the tests previously here, I see this level of verbosity in the code for function names